### DISCUSSION TITLE
How can I detect serverless function's timeout in my Node.js project

### DISCUSSION THREAD

**Discussion by Tupiet** [vercel/community#882] [Help] [answered]
Hey! Some months ago, I deployed a Node.js project to Vercel, using the serverless functions. The project consists of an API (express.js), which returns me some data generated randomly. The user makes a request to this API and send a JSON, which contains the type of data (names, emails, numbers, etc.) and the quantity of data. And here's the problem: as I'm in the hobby plan, the timeout for the serverless functions is about 10 seconds. Depending on the quantity, the timeout is exceeded or not, firing a 504 error. 

I'd like the API to send a message if the timeout was exceeded, but I don't know how I can do it. This is the code of the API, which gets the body of the request (the data I want to receive, with JSON format), then that JSON is managed, and finally I send another JSON with the data:

```
app.post('/api', (req, res) => {
    const body = req.body

    let result = manageData(body)
    res.send(result)
})
```

Is there any way I could detect if I'm exceeding the timeout, and then send a message explaining the problem?

Hope anyone can help me, and thanks for reading this!

If you need more data, please tell me. 

 Have a nice day.

**Comment by swarnava** [comment, marked as answer]
Hi,

If you have setup Logdrain with your application, you can create an alert when execution reach to certain threshold. Learn more here: https://vercel.com/support/articles/how-do-i-store-logs-on-vercel

**Reply by Tupiet** [reply]
Hey, thanks for your fast anwer. Please correct me if I'm wrong, but those logs would send me an email if I exceeded the timeout, right? If yes, then probably I haven't explained myself correctly.

I would need to send an API message telling that the timeout was exceeded, but not sending myself a mail. Sorry for the confusion, and hope this makes things clearer. Thanks!

**Reply by swarnava** [reply]
You can add more logging or delay using set timeout method and send it to logdrain: https://vercel.com/guides/what-can-i-do-about-vercel-serverless-functions-timing-out#check-for-upstream-errors
